src/common/helpers/poller.js

```diff
@@ -0,0 +1,11 @@
+
+// Extracts unique repository URLs from a given (parsed) snapcraft.yaml
+export function extractPartsToPoll(snapcraft_yaml) {
+  const parts = Object.values(snapcraft_yaml.parts || {});
+  const gh_parts = parts.filter(function (e) {
+    return (e['source-type'] == 'git' &&
```

Comment:
`source-type` isn't always set. Snapcraft does its best to [guess in that situation](https://github.com/snapcore/snapcraft/blob/1fdc9b39d96ffd043911a4586bd90c7cfb5e07b0/snapcraft/internal/sources/__init__.py#L176), doing so correctly if the ends in `.git`, for example.